tests/fixture/issue-2548/case2/output/index.js

```diff
@@ -8,10 +8,17 @@ Object.defineProperty(exports, "X", {
         return _z.default;
     }
 });
-var _exportNames = {
```

Comment:
I'm afraid the current PR may actually regress issue #2548 in this state.

Here, _exportNames is removed by this PR in this case.  Which means, later on we will may an `Object.defineProperty()` for `X`, if `X` is exported by `Z` using `export * from "./Z"`.  This will result in multiple calls to `Object.defineProperty` for the same property.